### PR TITLE
Test: Add snapshot test for changelog formatting

### PR DESCRIPTION
**Pull request by gziolo**
## Description

A follow-up to my comment https://github.com/WordPress/gutenberg/pull/34042#issuecomment-897803603:

> I'm tempted to add unit tests so we don't have to test changes manually for every simple modification that CI can verify.
> 
> From what I see, we can extract everything after `await fetchAllPullRequests( octokit, settings );` call to a method that could be used to test how entries are grouped and further processed.

The idea is very simple. I executed:

```bash
npm run changelog -- --milestone="Gutenberg 11.3" 
```

and saved the result from `fetchAllPullRequests` call as JSON.

I extracted all processing logic to `formatChangelog` function. The test uses the saved fixture (quite large) to verify the generated output string using the stored snapshot that should be good enough for visual verification.

## How has this been tested?

```bash
npm run test-unit bin/plugin/commands/test/changelog.js  
```

## Screenshots 

Screen Shot 2021-08-12 at 19 36 22

Despite the huge size of the fixture with the response from GitHub it seems to be pretty fast.

## Types of changes

New test.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: bin/plugin/commands/test/__snapshots__/changelog.js.snap [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+#### Document Settings
+- Disable spellcheck and autocomplete in permalink slug field. ([33708](https://github.com/WordPress/gutenberg/pull/33708))
+
+#### Template Editor
+- Template Mode: Add busy state to template creation modal. ([33076](https://github.com/WordPress/gutenberg/pull/33076))
+
+#### Global Styles
+- Dimensions Panel: Add new ToolsPanel component and update spacing supports. ([32392](https://github.com/WordPress/gutenberg/pull/32392))
+
+#### Uncategorized
```
Comment (getdave) [inline bin/plugin/commands/test/__snapshots__/changelog.js.snap]: This is out of date with `trunk` as of https://github.com/WordPress/gutenberg/pull/34037

File: bin/plugin/commands/test/__snapshots__/changelog.js.snap [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+- Editor: Safer isPreviewingPost check. ([33840](https://github.com/WordPress/gutenberg/pull/33840))
+
+#### Meta Boxes
+- Fix Safari 13 metaboxes from overlapping the content. ([33817](https://github.com/WordPress/gutenberg/pull/33817))
+
+#### Build Tooling
+- Readable JS assets Plugin: Fix webpack 5 support. ([33785](https://github.com/WordPress/gutenberg/pull/33785))
+
+#### Documentation
+- fix: Broken link in documentation to block support mechanism. ([33780](https://github.com/WordPress/gutenberg/pull/33780))
```
Comment (getdave) [inline bin/plugin/commands/test/__snapshots__/changelog.js.snap]: This usage of `Documentation` as a feature name will become out of date with `trunk` as soon as I merge https://github.com/WordPress/gutenberg/pull/34042. After that all `Documentation` tasks will group under the main `Documentation` section.

**Review by getdave — APPROVED** [review]
This is looking good and a great example of when using snapshots is actually helpful.

My only concern was whether you'd modified any of the code that builds the changelog as we're making quite a few changes in various PRs that are about to be merged into `trunk`. 

I wonder - by extracting the `formatChangelog` will you have lost some changes that have been merged into `trunk` since this PR was created? I noticed some inconsistencies so it makes me think that yes this will be the case.

I think (if you aren't already planning to do so) it would be good to copy/paste the changelog code from `trunk` into your `formatChangelog` once https://github.com/WordPress/gutenberg/pull/34042 has been merged. That will ensure everything is up to date and we don't lose any changes.

Thanks again for adding tests 🥇

**Comment by gziolo** [comment]
Thank you for your feedback @getdave. I will wait until #34042 lands and update the content of `formatChangelog` method. Thank you for catching it 👍🏻

**Comment by getdave** [comment]
> Thank you for your feedback @getdave. I will wait until #34042 lands and update the content of `formatChangelog` method. Thank you for catching it 👍🏻

Merged 👍

**Comment by getdave** [comment]
To be clear you should expect to see:

- Headings for feature sections not indented lists.
- All docs tasks under the main `Documentation` section heading and no "feature" documentation headings _anywhere_ else.
- No `Uncategorized` feature heading anywhere.
- All non-categoized items appear at the _top_ of each section, _without_ a heading, and _above_ the features.

**Comment by gziolo** [comment]
7dc5d34 - it looks like all the changes are applied correctly now. The diff in the snapshot is very simple to process and gives instant confirmation that the output looks as expected.

**Comment by getdave** [comment]
> [7dc5d34](https://github.com/WordPress/gutenberg/commit/7dc5d34b8c0b0eecb880c65d0ea465c87b6b3709) - it looks like all the changes are applied correctly now. The diff in the snapshot is very simple to process and gives instant confirmation that the output looks as expected.

Looks good to go 🚢

**Pull request merged**